src/Common/src/TypeSystem/Interop/MethodDesc.Interop.cs

```diff
@@ -63,11 +63,51 @@ public struct ParameterMetadata
         public bool HasFieldMarshal { get { return (_attributes & ParameterMetadataAttributes.HasFieldMarshal) == ParameterMetadataAttributes.HasFieldMarshal; } }
 
 
-        public ParameterMetadata(int index, ParameterMetadataAttributes attributes, TypeDesc marshalAs)
+        public ParameterMetadata(int index, ParameterMetadataAttributes attributes, MarshalAsDescriptor marshalAsDescriptor)
         {
             Index = index;
             _attributes = attributes;
-            MarshalAs = marshalAs;       
+            MarshalAsDescriptor = marshalAsDescriptor;
+        }
+    }
+
+    [Flags]
+    public enum NativeType : byte
```

Comment:
Will this also be used for field marshalling?

If so, maybe put it in a separate file, since it's not MethodDesc specific? (E.g. a file named MarshalAsDescriptor.cs in the same folder as this file).